README.md

```diff
@@ -0,0 +1,319 @@
+        // other builders
+        "store": "0.x"
+    }
+}
+```
+
+Configuring the store builder requires a `store` folder on the root of our app. If we want to create templates, we will need a `blocks.json` file inside that `store` folder. This means that, for now, our app's folder structure contains at least:
+
+```txt
+.
+├── manifest.json
+└── store
+    └── blocks.json
+```
+
+The Dream Store has already defined a home page template, which is the one used by default. Let's take a look at what that portion of the `blocks.json` from the `vtex.dreamstore` app looks like.
+
+```jsonc
+{
+    "store/home": {
+        "name": "Dream Store home page",
+        "blocks": ["header/full", "banner", "shelf", "footer"],
+    },
+    // some other blocks
+}
+```
+
+This file is declaring a `store/home` block which has a descriptive `name` which will be displayed in the Storefront editor. It also declares which blocks are contained within it. This means that the same `blocks.json` may declare those blocks or that the app depends on apps that declare those blocks inside their `blocks.json` files. In this specific case, the `vtex.dreamstore` app depends on apps that declare each of those, `vtex.header` declaring `header`, `vtex.banner` declaring `banner`, `vtex.shelf` declaring `shelf` and `vtex.footer` declaring `footer`.
+
+Why did Dream Store use `store/home` instead of just calling the block `home`? This means that `store/home` is a block that **extends** a `store` block. The `vtex.dreamstore` app depends on `vtex.store` which declares the `store` block, let's take a look at that declaration.
+
+```jsonc
+{
+    "store": {
+        "name": "Base store block",
+        "block": ["header", "footer"]
+    },
+    // other blocks
+}
+```
+
+It actually has some other definitions, but this part means that every extension of a `store` block contains a `header` and a `footer`. If `store/home` didn't contain either a `header` or a `footer` it would fail to build.
+
+Note that `store/home` could use `header/full` instead of `header`, even though `store` forced it to have a `header` block. That is because the main advantage of extensibility is that **every block can be replaced by an extension of it**. Meaning that, since `header/full` extends `header`, we could use it to fill a gap destined for `header`.
+
+To create a block that can replace the `store/home` via the Storefront editor, we need only to create an extension of `store/home`. Let's do that using our `blocks.json` file within the `partner.theme` app.
+
+```jsonc
+{
+    "store/home/alternative": {
+        "name": "Alternative home page",
+        "description": "A home page with a shelf and two banners",
+        "blocks": ["header/full", "banner", "shelf", "banner", "footer"]
+    }
+}
+```
+
+Great! But what does `store/home` mean? What about `banner`? Well, we need to depend on the apps that declare them otherwise none of this makes any sense. Let's do that on our `manifest.json`.
```

Comment:
If we're using a **store** builder, I think we should enforce the dependency on `store/*`, just like the react builder injects `render` and other stuff (`tachyons`, `react-dom`, `react`, etc). WDYT?

Reply:
We've talked about doing that, but there are some problems:
1) Does this mean that every base block definition created by us are in `vtex.store`? Isn't it useful to have it spread across various apps? What if we change the contract for `shelf` and `shelf` is defined in `vtex.store`? Will we need to change the major for `vtex.store`? I don't like this.
2) If the base definitions are not in `vtex.store`, there are some apps that can't depend in `vtex.store`, such as `vtex.shelf`, because `vtex.store` already depends on `vtex.shelf`. That would create a cycle. Do we only allow `vtex` apps not to extend `vtex.store`?